docs/formats/zst.md

```diff
@@ -190,128 +117,132 @@ independently of each other, similar to zng compression framing.
+from column streams, i.e., to map columns back to composite values.
 
 > Of course, the reassembly section also provides the ability to extract just subsets of columns
 > to be read and searched efficiently without ever needing to reconstruct
 > the original rows.  How performant this is all done is up to any particular
-> zst implementation.
+> ZST implementation.
 
 > Also, the reassembly section is in generally vastly smaller than the data section
 > so the goal here isn't to express information in cute and obscure compact forms
 > but rather to represent data in easy-to-digest, programmer-friendly form that
-> leverages zng.
+> leverages ZNG.
 
-The reassembly section is a zng stream.  Unlike parquet,
+The reassembly section is a ZNG stream.  Unlike Parquet,
 which uses an externally described schema
 (via [Thrift](https://thrift.apache.org/)) to describe
-analogous data structures, we simply reuse zng here.
+analogous data structures, we simply reuse ZNG here.
 
-> So, we are using zng to encode zng in column format.  Yo dawg.
+#### The Super Types
 
-#### The Schema Definitions
+This reassembly stream encodes 2*N+1 Zed values, where N is equal to the number
+of top-level Zed types that are present in the encoded input.
+To simplify terminology, we call a top-level Zed type a "super type",
+e.g., there are N unique super types encoded in the ZST object.
 
-This reassembly zng stream encodes 2*N+1 zng records, where N is equal to the number
-of top-level zng record types that are present in the encoded input.
-To simplify terminology, we call a top-level zng record type a "schema",
-e.g., there are N unique schemas encoded in the zst object.
+These N super types are defined by the first N values of the reassembly stream
+and are encoded as a null value of the indicated super type.
+A super type's integer position in this sequence defines its identifier
+encoded in the super column (defined below).  This identifier is called
+the super ID.
 
-Each of these N schemas gets defined as a record value, comprised
-of unset fields appearing in the same order
-as they were encountered in the original zng stream.
-In this way, a fresh zng type context can be created to read these N records and
-that type context will have precisely the same structure as the type context
-from the original zng stream.  This is guaranteed by the deterministic nature
-of embedded zng type definitions.
+> Change the first N values to type values instead of nulls?
 
-The next N+1 records contain reassembly information for each of the N schemas
-where each record is used to create column streams to reconstruct the original
-zng records.  The schemas do not overlap as columns from a record from any one
-schema are not intermixed with columns of another.
+The next N+1 records contain reassembly information for each of the N super types
+where each record defines the column streams needed to reconstruct the original
+Zed values.
 
 #### Segment Maps
```

Comment:
This section is describing the 2N+1 records so moving the description of the next N+1 records would break the flow.  This section is describing how you bootstrap to get the root record types.  The next section tells you how to decode each type that you encounter from the root.

I agree all this can be improved but this isn't the right direction.